src/client/src/region.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use api::v1::region::{region_request, RegionRequest, RegionResponse};
+use api::v1::{AuthHeader, RequestHeader, ResponseHeader};
+use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+use common_error::status_code::StatusCode;
+use common_telemetry::timer;
+use derive_builder::Builder;
+use snafu::OptionExt;
+
+use crate::error::{IllegalDatabaseResponseSnafu, Result, ServerSnafu};
+use crate::{metrics, Client};
+
+type AffectedRows = u64;
+
+#[derive(Debug, Builder)]
+pub struct RegionClient {
+    #[builder(setter(into), default = "DEFAULT_CATALOG_NAME.to_string()")]
+    catalog: String,
+
+    #[builder(setter(into), default = "DEFAULT_SCHEMA_NAME.to_string()")]
+    schema: String,
+
+    #[builder(default)]
+    authorization: Option<AuthHeader>,
+
+    #[builder(setter(into), default)]
+    dbname: String,
+
+    #[builder(setter(strip_option), default)]
+    trace_id: Option<u64>,
+
+    #[builder(setter(strip_option), default)]
+    span_id: Option<u64>,
```

Comment:
FWIW, this is not suitable to be a field. This is misleading. How about putting these in fn param as they should change for each fn call?

Reply:
Add a todo here for myself, will resolve it when we have them.